controller/validation.go

```diff
@@ -0,0 +1,45 @@
+package controller
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/expenseledger/web-service/service"
+	"github.com/gin-gonic/gin"
+	"golang.org/x/net/context"
+)
+
+// ValidateHeader validate header
+func ValidateHeader() gin.HandlerFunc {
+	firebase, err := service.GetFirebaseInstance()
+
+	return func(c *gin.Context) {
+		if err != nil {
+			buildAbortContext(c, fmt.Errorf("Cannot initialize firebase, %v", err), http.StatusInternalServerError)
+			return
+		}
+
+		token := c.Request.Header.Get("X-Token")
```

Comment:
I encourage you to use `authorization` instead. Not sure. Could you do some research on that?